py3status/modules/wwan_status_nm.py

```diff
@@ -0,0 +1,283 @@
+    format_up = 'WWAN: {status} - {operator} {netgen} ({signal}%) {format_ipv4}'
+    modem = None
+
+    def post_config_hook(self):
+        # network states dict
+        # https://www.freedesktop.org/software/ModemManager/api/1.0.0/ModemManager-Flags-and-Enumerations.html#MMModemState
+        self.states = {
+            -1: 'failed',
+            0: 'unknown',
+            1: 'initializing',
+            2: 'locked',
+            3: 'disabled',
+            4: 'disabling',
+            5: 'enabling',
+            6: 'enabled',
+            7: 'searching',
+            8: 'registered',
+            9: 'disconnecting',
+            10: 'connecting',
+            11: 'connected'
+        }
+
+        # network speed dict
+        # https://www.freedesktop.org/software/ModemManager/api/1.0.0/ModemManager-Flags-and-Enumerations.html#MMModemAccessTechnology
+        self.speed = {
+            0: STRING_UNKNOWN,
+            1 << 0: 'POTS',
+            1 << 1: 'GSM',
+            1 << 2: 'GSM Compact',
+            1 << 3: 'GPRS',
+            1 << 4: 'EDGE',
+            1 << 5: 'UMTS',
+            1 << 6: 'HSDPA',
+            1 << 7: 'HSUPA',
+            1 << 8: 'HSPA',
+            1 << 9: 'HSPA+',
+            1 << 10: '1XRTT',
+            1 << 11: 'EVDO0',
+            1 << 12: 'EVDOA',
+            1 << 13: 'EVDOB',
+            1 << 14: 'LTE'
+        }
+
+        # network registration states
+        # https://www.freedesktop.org/software/ModemManager/api/1.0.0/ModemManager-Flags-and-Enumerations.html#MMModem3gppRegistrationState
+        self.rstates = {
+            0: 'IDLE',
+            1: 'HOME',
+            2: 'SEARCHING',
+            3: 'DENIED',
+            4: 'UNKNOWN',
+            5: 'ROAMING'
+        }
+
+        self.bus = SystemBus()
+
+    def wwan_status_nm(self):
+        data = {}
+        full_text = self.py3.safe_format(self.format_error,
+                                         dict(
+                                             error=STRING_NO_MODEM,
+                                             status="Disconnected"))
+        color = self.py3.COLOR_BAD
+
+        try:
+            modemmanager_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS)
+            modems = modemmanager_proxy.GetManagedObjects()
+        except:
+            color = self.py3.COLOR_BAD
+            full_text = self.py3.safe_format(self.format_error, data)
+            modems = {}
+
+        # browse modems objects
+        for objects in modems.items():
+            modem_path = objects[0]
+            modem_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS, modem_path)
+
+            # we can maybe choose another selector
+            eqid = str(modem_proxy.EquipmentIdentifier)
+
+            # use selected modem or first find
+            if self.modem is None or self.modem == eqid:
+
+                try:
+                    # get status informations
+                    status = modem_proxy.GetStatus()
```

Comment:
Does this help with simplifying things so we don't have to do some things below?
```
modem_proxy.GetStatus

modem_data = {}
for key, value in modem_proxy.GetStatus().items():
    key = key.lower().replace('-', '_')
    if isinstance(value, [list, tuple])
        if len(value) > 1:
            for i, v in enumerate(value):
                modem_data['%s_%s' % (key, i)] = v
        else:
            modem_data[key] = v
    else:
        modem_data[key] = v
```
With this, the new placeholders should look like: `signal_quality_0`, `signal_quality_1`, `access_technologies`, `m3gpp_registration_state`, `m3gpp_operator_name`, etc.

Reply:
Hmm, i will study this, i tried to do this before but my version bug and signal-quality was never replace

Reply:
@cyrinux Try again. My code is shobby and untested.